network/src/protocols/wire/messaging/v1/mod.rs

```diff
@@ -26,16 +25,15 @@ pub enum NetworkMessage {
 
 /// Unique identifier associated with each application protocol.
 /// New application protocols can be added without bumping up the MessagingProtocolVersion.
-#[repr(u8)]
-#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq, Deserialize_repr, Serialize_repr)]
+#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq, Deserialize, Serialize)]
```

Comment:
Do we want to eliminate the specificity here? One advantage of the labeling is that even if the ordering changes, the values are kept constant.

Reply:
@matbd did check with me on this. However, I think keeping specificity is good, not just if things are re-ordered, but also if/when they get removed. Thanks for pointing out, @davidiw!

Reply:
I don't mind reverting but let me mention a couple of things that motivated me:
* Other enums in this file do not use `Serialize_repr`.
* LCS/Serde is strict so if one removes a case in the future (say delete the line `HealthCheckerRpc = 5`) then old messages will fail at parsing time. (Do we ever want this?)
* I am building some tooling to record the Serde format of interesting types, and detect accidental changes over time. (See https://github.com/libra/libra/pull/3030 for an example)